src/containers/PackingOverview/PackingPage/PackingPage.js

```diff
@@ -124,6 +130,24 @@ export default (class PackPage extends Component {
     return;
   };
 
+  handleShoppingCart = async (index, e) => {
+    const list_id = await findOrCreateShoppingCart(
+      index,
+      this.state,
+      this.props.lists
+    );
+    const result = await addToShoppingCart(index, this.state, list_id);
+    const { newState, updateParent } = result;
+    if (updateParent) {
+      this.props.updateLists();
+      if (newState) this.setState(newState);
+      return;
+    } else {
+      if (newState) this.setState(newState);
+      return;
+    }
```

Comment:
A few nitpicky thoughts on clean code:

* There's no need to `return` at the end of these two blocks (the `if` and the `else`), as they're at the end of the function
* If an `if` statement ends with a `return`, there's no need to use an `else`, as that code won't be reached anyway
* Your `i (newState) this.setState(newState)` line is duplicated at the end of both blocks; you could put it once afterwards if you remove the `return;`s